src/inmanta/agent/handler.py

```diff
@@ -470,19 +490,45 @@ def run() -> None:
 
         return f.result()
 
-    def set_cache(self, cache: AgentCache) -> None:
-        self.cache = cache
-
     def get_client(self) -> protocol.SessionClient:
         """
         Get the client instance that identifies itself with the agent session.
-
         :return: A client that is associated with the session of the agent that executes this handler.
         """
         if self._client is None:
             self._client = protocol.SessionClient("agent", self._agent.sessionid)
         return self._client
 
+
+@stable_api
+class ResourceHandler(HandlerABC):
```

Comment:
```suggestion
class ResourceHandler(HandlerABC[R]):
```
I think. Then update all type annotations of `resources.Resource` to `R` instead. Same for the `CRUDHandler`.